### PR TITLE
[DR-1393] Allow upgrades on webapp

### PR DESCRIPTION
**Pull request by alegarciamina**
Hi @andresmoschini ,
I've created this PR to start reviewing the code while continue working.

Changes:

- Show  "change plan" for free or paid users, but not when there are scheduled agreements 
- Only show DragMe if it is a free trial 
- Do not allow to choose the same plan that user already has (same plan is defined by comparing includedDeliveries and currentIpsPlanCount values between plan and current agreement) 
- Do not allow to choose plans cheaper than current plan, show a Contact Us button.
- Show billing information pre-loaded if exists
- Set initial value for slider based on current plan (defaultPlanDeliveries).

### CODE REVIEW

**Comment by andresmoschini** [comment]
@alegarciamina please re-base on `develop` branch, I am not sure but maybe my changes in _limits loading_ could collide with come of ours.

**Comment by andresmoschini** [comment]
[**@alegarciamina**](https://github.com/alegarciamina) commented on [17 de oct. de 2017 17:41 ART](https://github.com/DopplerRelay/relay-webapp/pull/138#issue-266271176 "2017-10-17T20:41:13Z - Replied by Github Reply Comments"):
> - Set default email per month value

I did not understand.

**Comment by alegarciamina** [comment]
@andresmoschini, by "Set default email per month value" I mean to set a default value of the slider (currently 60k) based on current plan. This was already done by Mauro on commit [0bc89c1](https://github.com/DopplerRelay/relay-webapp/pull/131/commits/0bc89c15a4df6b462f1654924bf1f831fed084e0) and I was planning to implement it.

2017-10-18_09h23_54

**Comment by andresmoschini** [comment]
[**@alegarciamina**](https://github.com/alegarciamina) commented on [18 de oct. de 2017 09:24 ART](https://github.com/DopplerRelay/relay-webapp/pull/138#issuecomment-337573181 "2017-10-18T12:24:28Z - Replied by Github Reply Comments"):
> ... by "Set default email per month value" I mean to set a default value of the slider (currently 60k) based on current plan ...

Oh, I understand now, thanks.

File: src/wwwroot/controllers/PlanCtrl.js [hunk 1]
```diff
@@ -44,6 +44,10 @@
         vm.currency = response.data.currency;
         vm.isFreeTrial = response.data.fee && response.data.includedDeliveries ? false : true;
         vm.currentIpsPlanCount = response.data.ips_count || 0;
+        vm.isUpdatePlanAllowed = response.data.endDate ? false : true;
+        if (!vm.isFreeTrial){
+           vm.hideDragMe = true;
+        }
```
Comment (andresmoschini) [inline src/wwwroot/controllers/PlanCtrl.js]: I would rewrite it as:

```js
vm.isUpdatePlanAllowed = !response.data.endDate;
vm.hideDragMe = !vm.isFreeTrial;
```

File: src/wwwroot/partials/settings/my-plan.html [hunk 2]
```diff
@@ -27,7 +27,7 @@
         <spinner class="spinner" ng-class="vm.planInfoLoader ?  'show':'hide'"></spinner>
       </div>
       <div class="flex one flex--wrap">
-        <button ng-show="vm.isFreeTrial" class="button button--small" ng-class="vm.pricingChartDisplayed ? 'button--disabled' : 'active'" ng-click="vm.showPricingChart()">{{ 'change_plan_text' | translate }}</button>
+        <button ng-show="vm.isUpdatePlanAllowed" class="button button--small" ng-class="vm.pricingChartDisplayed ? 'button--disabled' : 'active'" ng-click="vm.showPricingChart()">{{ 'change_plan_text' | translate }}</button>
```
Comment (andresmoschini) [inline src/wwwroot/partials/settings/my-plan.html]: I would rename `isUpdatePlanAllowed` as `isChangePlanAllowed` or `allowChangePlan`

**Comment by andresmoschini** [comment]
Nice work @alegarciamina! there were less changes that I expected (that is good).

File: src/wwwroot/controllers/BillingCtrl.js [hunk 3]
```diff
@@ -59,6 +59,29 @@
         }
         vm.currentCurrency = planSelected.currency;
         vm.planPrice = planSelected.fee + (planSelected.ips_count * planSelected.cost_by_ip || 0);
+
+        return settings.getCurrentPlanInfo().then(function(response) {
```
Comment (andresmoschini) [inline src/wwwroot/controllers/BillingCtrl.js]: This information should be already available, right?

Comment (andresmoschini) [reply src/wwwroot/controllers/BillingCtrl.js]: Oh, no, it is another page / controller.

**Comment by andresmoschini** [comment]
Nice tests @alegarciamina !

**Pull request merged**